### PR TITLE
CIRC-4064 - add activity tracking and latest values to the irondb tag…

### PR DESCRIPTION
**Pull request by bdunavant**
… search page.  We need to put something in on how to enable the latest-value code, so I can link to it.

### CODE REVIEW

**Comment by bdunavant** [comment]
@postwait I need to understand how this is turned on/off so that can be documented.

File: configuration.md [hunk 1]
```diff
@@ -555,6 +558,31 @@ IRONdb starts.
 
 Default: The number of physical processor cores in the system.
 
+#### surrogate_database activity_tracking
+
+Directs IRONdb as to if it should maintain activity windows for metrics.
+
+Default: false
```
Comment (postwait) [inline configuration.md]: Default: true

File: configuration.md [hunk 2]
```diff
@@ -555,6 +558,31 @@ IRONdb starts.
...
+
+#### surrogate_database latest_future_bound
+
+Forward looking "cap" on if a metric will be considered for the latest value.  By
+default if a metric timestamp is more than 4 hours in the future, it will be ignored
+for consideration as a replacement for the lastest value.  These values are only
+updated at ingestion time.
+
+This value can be from 0 (ignore any future timestamps) to 4h (maximum).
```
Comment (postwait) [inline configuration.md]: `0s`  I *think* the parameters must be time spans.

File: configuration.md [hunk 3]
```diff
@@ -555,6 +558,31 @@ IRONdb starts.
 
 Default: The number of physical processor cores in the system.
 
+#### surrogate_database activity_tracking
+
```
Comment (esproul) [inline configuration.md]: This paragraph and the next one: "Directs IRONdb as to if it should" is... rather tortured grammar.  Elsewhere we've maintained an active voice, so perhaps "Maintain activity windows for metrics", and "Maintain latest values...".

File: configuration.md [hunk 4]
```diff
@@ -555,6 +558,31 @@ IRONdb starts.
...
+Directs IRONdb as to if it should maintain latest values for all newly seen metrics values
+during ingestion.  If set to false, it will only maintain lastest values for metrics that 
+have been specifically "asked for." (see (Searching)[/api/search-tags.html])
+
+Default: false
+
+#### surrogate_database latest_future_bound
+
+Forward looking "cap" on if a metric will be considered for the latest value.  By
```
Comment (esproul) [inline configuration.md]: How about "Upper bound on whether..." ?

File: api/search-tags.md [hunk 5]
```diff
@@ -20,12 +20,16 @@ GET
  * `query`               : See [Tag Queries](/tag-queries.md) for more info on tag queries.
  * `activity_start_secs` : (optional) The start time from which to pull data, represented in seconds since the unix epoch.
  * `activity_end_secs`   : (optional) The end time up to which data is pulled, represented in seconds since the unix epoch.
+ * `latest`              : (optional, default 0) Specify if the latest values for the metric should be returned.  Parameters:
```
Comment (esproul) [inline api/search-tags.md]: I find this description confusing. I could read it as "the most recent value won't be included in the result set unless I set this", as opposed to what I know it is, "return the latest value as a separate object".  I don't have a good suggestion here; the terminology is just confusing.

Comment (bdunavant) [reply api/search-tags.md]: Huh?  your initial reading is accurate.  The latest values will not be included in the result set unless you set it.

Comment (esproul) [reply api/search-tags.md]: OK, then this is MY confusion. I realize now this section is talking about tag searching, so we're not pulling actual metric values anyway.

One question though, is it "values" (plural) or exactly one "latest value"?

Comment (bdunavant) [reply api/search-tags.md]: Last 2 values.

**Pull request merged**